Parse/src/main/java/com/parse/ParseRESTQueryCommand.java

```diff
@@ -41,38 +50,41 @@
     HashMap<String, String> parameters = new HashMap<>();
     List<String> order = state.order();
     if (!order.isEmpty()) {
-      parameters.put("order", ParseTextUtils.join(",", order));
+      parameters.put(KEY_ORDER, ParseTextUtils.join(",", order));
     }
 
     ParseQuery.QueryConstraints conditions = state.constraints();
     if (!conditions.isEmpty()) {
       JSONObject encodedConditions =
           (JSONObject) encoder.encode(conditions);
-      parameters.put("where", encodedConditions.toString());
+      parameters.put(KEY_WHERE, encodedConditions.toString());
     }
 
     // This is nullable since we allow unset selectedKeys as well as no selectedKeys
     Set<String> selectedKeys = state.selectedKeys();
     if (selectedKeys != null) {
-      parameters.put("keys", ParseTextUtils.join(",", selectedKeys));
+      parameters.put(KEY_KEYS, ParseTextUtils.join(",", selectedKeys));
     }
 
     Set<String> includeds = state.includes();
     if (!includeds.isEmpty()) {
-      parameters.put("include", ParseTextUtils.join(",", includeds));
+      parameters.put(KEY_INCLUDE, ParseTextUtils.join(",", includeds));
+    }
+
+    // Respect what the caller wanted for limit, even if count is true, because
+    // parse-server supports it. Currently with our APIs, when counting, limit will always be 0,
+    // but that logic is in ParseQuery class and we should not do that again here.
+    int limit = state.limit();
+    if (limit >= 0) {
+      parameters.put(KEY_LIMIT, Integer.toString(limit));
     }
```

Comment:
Your link is a good reference because it says

`With a nonzero limit, that request would return results as well as the count.`

So `parse-server` supports counting and finding altogether. We don’t currently support this with our exposed APIs, but we might want one day in the future.
So since `parse-server` supports it and `ParseRESTQueryCommand` is a low level interface to reach the API, I think it should allow any limit.
The important thing is that any call to `ParseQuery.count()` will use `limit = 0`. And that is already true https://github.com/parse-community/Parse-SDK-Android/blob/master/Parse/src/main/java/com/parse/ParseQuery.java#L1300 : 

`final State<T> state = copy.setLimit(0).build();`

So the logic `if (count) limit = 0` is already in the `count()` methods. The issue was that `ParseRESTQueryCommand` was ignoring `limit = 0` imposed by `ParseQuery`.